tests/validators/test_typed_dict.py

```diff
@@ -1092,3 +1092,109 @@ def test_bad_default_factory(default_factory, error_message):
     )
     with pytest.raises(TypeError, match=re.escape(error_message)):
         v.validate_python({})
+
+
+class TestOnError:
+    def test_on_error_bad_name(self):
+        with pytest.raises(SchemaError, match="Value must be one of: 'raise', 'omit', 'fallback_on_default'"):
+            SchemaValidator({'type': 'typed-dict', 'fields': {'x': {'schema': {'type': 'str'}, 'on_error': 'rais'}}})
+
+    def test_on_error_bad_omit(self):
+        with pytest.raises(SchemaError, match='Field "x": "on_error = omit" cannot be set for required fields'):
+            SchemaValidator({'type': 'typed-dict', 'fields': {'x': {'schema': {'type': 'str'}, 'on_error': 'omit'}}})
+
+    def test_on_error_bad_fallback_on_default(self):
+        with pytest.raises(
+            SchemaError, match='Field "x": "on_error = fallback_on_default" requires a `default` or `default_factory`'
+        ):
+            SchemaValidator(
+                {'type': 'typed-dict', 'fields': {'x': {'schema': {'type': 'str'}, 'on_error': 'fallback_on_default'}}}
+            )
+
+    def test_on_error_raise_by_default(self, py_and_json: PyAndJson):
+        v = py_and_json({'type': 'typed-dict', 'fields': {'x': {'schema': {'type': 'str'}}}})
+        assert v.validate_test({'x': 'foo'}) == {'x': 'foo'}
+        with pytest.raises(ValidationError):
+            v.validate_test({'x': ['foo']})
+
+    def test_on_error_raise_explicit(self, py_and_json: PyAndJson):
+        v = py_and_json({'type': 'typed-dict', 'fields': {'x': {'schema': {'type': 'str'}, 'on_error': 'raise'}}})
+        assert v.validate_test({'x': 'foo'}) == {'x': 'foo'}
+        with pytest.raises(ValidationError):
```

Comment:
can you put at least `matches='[kind=...,]` so we're clear what error we're expecting.